### PR TITLE
Bump various dependency versions

### PR DESCRIPTION
**Pull request by Ekleog-NEAR**
In particular:
- cpufeatures
- ahash
- hashbrown
- criterion
- pretty-assertions
- tracing-subscriber
- chrono
- h2
- tempfile
- hermit-abi

Also:
- replaced ansi_term with yansi
- removed our dependency on atty altogether

All this together, alongside with https://github.com/near/nearcore/pull/10201, gets us to zero cargo-audit warnings :)

This might be best reviewed commit-by-commit, to more easily see which changes are due to which dependency (though I tried bumping chrono and replacing ansi_term at the same time, resulting in alternating commits as my test compilations made progress)

### CODE REVIEW

**Comment by wacban** [comment]
> All this together, alongside with https://github.com/near/nearcore/pull/10201, gets us to zero cargo-audit warnings :)

❤️

File: chain/client-primitives/src/types.rs [hunk 1]
```diff
@@ -216,7 +216,7 @@ pub fn format_shard_sync_phase(
     match &shard_sync_download.status {
         ShardSyncStatus::StateDownloadHeader => format!(
             "{} requests sent {}, last target {:?}",
-            paint("HEADER", Purple.bold(), use_colour),
+            paint("HEADER", Magenta.style().bold(), use_colour),
```
Comment (wacban) [inline chain/client-primitives/src/types.rs:219]: lol, did they actually rename the color when bumping version or do you just like megenta more? :)

Comment (wacban) [reply chain/client-primitives/src/types.rs:219]: ah nevemind, it's different library

File: chain/client/src/info.rs [hunk 2]
```diff
@@ -538,9 +538,9 @@ impl InfoHelper {
 
     fn log_chain_processing_info(&mut self, client: &crate::Client, epoch_id: &EpochId) {
         let chain = &client.chain;
-        let use_colour = matches!(self.log_summary_style, LogSummaryStyle::Colored);
+        let use_color = matches!(self.log_summary_style, LogSummaryStyle::Colored);
```
Comment (wacban) [inline chain/client/src/info.rs:541]: As a British resident I strongly disagree ;)

File: core/o11y/src/subscriber.rs [hunk 3]
```diff
@@ -93,8 +93,8 @@ pub enum ColorOutput {
 }
 
 fn is_terminal() -> bool {
-    // Crate `atty` provides a platform-independent way of checking whether the output is a tty.
-    atty::is(atty::Stream::Stderr)
+    use std::io::IsTerminal;
+    std::io::stderr().is_terminal()
```
Comment (wacban) [inline core/o11y/src/subscriber.rs:97]: nice

**Review by wacban — APPROVED** [review]
LGTM

File: chain/client/src/info.rs (hunk 2 above)
Comment (wacban) [reply chain/client/src/info.rs:541]: (I'm fine with it, just don't tell the king)

Comment (Ekleog-NEAR) [reply chain/client/src/info.rs:541]: Heh 😅 turns out `ansi_term` was written by british people, and `yansi` by americans, or so it seems :p

**Pull request merged**